### PR TITLE
Closes #27, Closes #28: Updated the GeometryNormalizer to support dyn…

### LINKED ISSUE #27: Add GeoWave JexlQueryFunctions to enable Lucene syntax for geo queries

### LINKED ISSUE #28: Update GeometryNormalizer to support normalization of multiple string geometry formats.
At the moment, we only support WKT.  We should update the geometry normalizer so that it can handle additional geometry representations besides WKT.

### PR DESCRIPTION
**Pull request by jwomeara**
…amic geometry string parsing.

Also, added GeoWave JexlQueryFunctions in order to enable support for Lucene formatted GeoWave queries.

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/attributes/Geometry.java [hunk 1]
```diff
@@ -34,7 +33,7 @@ protected Geometry() {
     
     public Geometry(String wellKnownText, Key docKey, boolean toKeep) {
         super(docKey, toKeep);
-        setGeometryFromWkt(wellKnownText);
+        setGeometryFromGeoString(wellKnownText);
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/attributes/Geometry.java]: change variable name

File: warehouse/query-core/src/main/java/datawave/query/language/functions/jexl/Geowave.java [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+import datawave.webservice.query.exception.BadRequestQueryException;
+import datawave.webservice.query.exception.DatawaveErrorCode;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+
+public interface Geowave {
+    
+    class Contains extends JexlQueryFunction {
```
Comment (ivakegg) [inline warehouse/query-core/src/main/java/datawave/query/language/functions/jexl/Geowave.java]: Lets consolidate some of this code: e.g. make an abstract class in the middle that gives us the validate and toString method implementations that are duplicated in each one of these.

File: web-services/client/src/main/java/datawave/data/normalizer/GeometryNormalizer.java [hunk 3]
```diff
@@ -2,29 +2,22 @@
...
-import mil.nga.giat.geowave.core.store.index.BasicIndexModel;
-import mil.nga.giat.geowave.core.store.index.CustomIdIndex;
-import mil.nga.giat.geowave.core.store.index.PrimaryIndex;
 import org.apache.commons.codec.binary.Hex;
-import org.apache.commons.lang.StringUtils;
 
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
+import java.util.*;
```
Comment (ivakegg) [inline web-services/client/src/main/java/datawave/data/normalizer/GeometryNormalizer.java]: By convention we have been expanding these.  We should enforce this somewhere of course so this is on us.  However you can change your IDE to force expansion.

File: warehouse/query-core/src/main/java/datawave/query/language/functions/jexl/Geowave.java (hunk 2 above)
Comment (tomnelson) [reply warehouse/query-core/src/main/java/datawave/query/language/functions/jexl/Geowave.java]: Can you add interface default methods for validate and toString so we don't need another class?

File: warehouse/query-core/src/main/java/datawave/query/attributes/Geometry.java (hunk 1 above)
Comment (jwomeara) [reply warehouse/query-core/src/main/java/datawave/query/attributes/Geometry.java]: Fixed!

File: warehouse/query-core/src/main/java/datawave/query/language/functions/jexl/Geowave.java (hunk 2 above)
Comment (jwomeara) [reply warehouse/query-core/src/main/java/datawave/query/language/functions/jexl/Geowave.java]: Fixed!

File: web-services/client/src/main/java/datawave/data/normalizer/GeometryNormalizer.java (hunk 3 above)
Comment (jwomeara) [reply web-services/client/src/main/java/datawave/data/normalizer/GeometryNormalizer.java]: D'oh!  Fixed!

**Pull request merged**